model/build/build.go

```diff
@@ -166,6 +166,31 @@ func UpdateActivation(buildIds []string, active bool, caller string) error {
 	return errors.Wrapf(err, "setting build activation to %t", active)
 }
 
+// UpdateActivationAndStatus updates builds with the given ids
+// to the given activation setting and marks builds as created if active.
+func UpdateActivationAndStatus(buildIds []string, active bool, caller string) error {
```

Comment:
We aren't actually using this function anywhere it looks like? But also if we intend to only use this in the case that all buildIds are being updated in a version, we could change this to take in the version ID and query on that.

(We could also maybe create a setTaskActivationForVersion and refactor setTaskActivationForBuilds so they can share a lot of code). I would note that SetTaskActivationForBuilds does call  UpdateVersionAndPatchStatusForBuilds, so I'm not sure where the refactor to improve this is that you mentioned in your comment

Reply:
i tried to refactor it and i think i try to minimize database queries as much as possible